shotover-proxy/src/transforms/load_balance.rs

```diff
@@ -118,7 +118,7 @@ mod test {
                 assert_eq!(guard.len(), 3);
                 for bc in guard.iter() {
                     let guard = bc.count.lock().await;
-                    assert_eq!(*guard, 30);
+                    assert_eq!(*guard, 0);
```

Comment:
This now seems like we have changed the behaviour of load balance? The count of each balanced chain should be 30 (90 ops split amongst three connections).

In load balance we are deliberately sharing the sender handles across multiple chains

Reply:
this could be solved with a get_handle method or something and using that in load_balance

Reply:
![image](https://user-images.githubusercontent.com/5120858/145136393-135fb66b-c1e5-4774-9852-12f7c7727f49.png)
I dont think the load balance implementation ever achieved something useful.
once self.active_connection is Some it can never be changed to another connection.

Reply:
This transform is also not exposed in our public API yet.

Reply:
I removed the assert from the test entirely:
* It would be confusing to have it `assert_eq(*guard, 0)` because that makes it seem like the correct behaviour
* Testing implementation details like this is not great

Reply:
That said I am also happy to just completely revert this 2nd commit and keep all the changes localized to the Tee transform.

Reply:
Ooooooh, is the idea behind ConnectionBalanceAndPool that it uses a new connection every time its cloned?

Reply:
I made this PR to document how ConnectionBalanceAndPool works https://github.com/shotover/shotover-proxy/pull/411

Reply:
Let's revert and pull this out for another PR -> Let's just fix the broken Tee / Auth workflow issue